src/distilabel/pipeline.py

```diff
@@ -402,10 +466,11 @@ def _build_dataset(  # noqa: C901
                                     raw_output=None,
                                     parsed_output=None,
                                 )
+                                for _ in range(num_outputs)
                             ]
                         )
             else:
-                processed_labels = batch_labels
+                processed_labels = labels
             labels = self._process_batch_labels(batch_labels=processed_labels)  # type: ignore
```

Comment:
IMO we can remove the `else`-statement and just `labels = self._process_batch_labels(batch_labels=processed_labels or labels) # type: ignore`